src/pudl/helpers.py

```diff
@@ -2301,7 +2301,7 @@ def get_parquet_table(
         )
 
     # Only enforce schema if we're reading all columns
-    if columns is None:
+    if set(columns) == set(resource.get_field_names()):
```

Comment:
slick!

Reply:
Seemed closer to what we *actually* mean.